nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -888,8 +899,25 @@ in {
 
       services.nginx.enable = mkDefault true;
 
-      services.nginx.virtualHosts.${cfg.hostName} = {
-        root = cfg.package;
+      services.nginx.virtualHosts.${cfg.hostName} = let
+        webroot = if cfg.config.overwriteWebroot == null then ""
+                  else if builtins.match "/.+[^/]" cfg.config.overwriteWebroot != null then cfg.config.overwriteWebroot
+                  else throw "nextcloud: invalid overwriteWebroot option";
+        webrootOrSlash = if webroot != "" then webroot else "/";
+        mkRoot = name: root: pkgs.stdenv.mkDerivation {
+          inherit name;
+          dontUnpack = true;
+          dontBuild = true;
+          dontFixup = true;
+          installPhase = ''
+            export target=$out${webroot}
+            mkdir -p $(dirname $target)
+            ln -s ${root} $target
+          '';
+        };
+        root = mkRoot "nextcloud-webroot" cfg.package;
+        approot = mkRoot "nextcloud-approot" cfg.home;
```

Comment:
This seems problematic: `cfg.home` is `root` inside `mkRoot` and thus it's apparently attempted to copy `/var/lib/nextcloud` into the store, correct? (because of `${root}` in the derivation). Also, how about simply using `pkgs.runCommand` here?

Reply:
`root` in `mkRoot` is a string, `mkRoot` should only create a symbolic link to `var/lib/nextcloud/`. I just checked on my server and that's indeed what's done.
Maybe this is a bit dirty, I did it because when using the `root` keyword in the nginx config, nginx is still searching for the entire url path inside that root. 
Maybe there is a way to tell nginx to remove the webroot from the path.

Reply:
Oof sorry, you're absolutely right!

Reply:
But what you want to achieve should be possible with `alias`: https://nginx.org/en/docs/http/ngx_http_core_module.html#alias